### ISSUE TITLE
Leading - before config "enable-xxx" options

### ISSUE THREAD

**Issue by beldmit** [openssl/openssl#13249]
I came across a weird behavior of the openssl configuration process.

When configuring via `./config -enable-weak-cipher` I got a segfault on O2/O3 optimization levels running tests. It's a human error passing arguments in such a way but it is can't be found out either by the `config` script or by a compiler. 

The --enable-xxx options are passed to the compiler as is and the build stops. I think it's worth providing some processing of such options (e.g. strip the leading minus) to avoid such behavior.

**Comment by levitte** [comment]
The basic design is that, apart from certain specific options we recognise, anything that starts with `-`, `+` or `/` is passed to the compiler.  This is a design choice in the Configure script that dates back to the beginning of time (or well, since afd1f9e80bb848bf1fa36ae3b9bfc51d63750525, which is almost at the beginning of time).

Point is that different compilers have so many different interesting options these days that it's impossible to predict what options will come through and what we should catch and possibly disqualify.  Can you guarantee that no compiler will ever accept `-enable-foo`?

So this does come down to users knowing their compilers, and your issues come down to user error.

Mind you, we could simply not pass configuration options to the compiler / linker, and require that users specify them as "make variable assignments" on the configure command line, just like you can with GNU's configure scripts, i.e. something like `./Configure CFLAGS='-O2 -g'`...  that would give us much bigger freedom to do a catch-all of configuration options that start with a dash.  The problem is all those examples on the net that will stop working.

**Comment by beldmit** [comment]
I agree with your analysis. So emitting a warning in such a case seems to be the best realistic approach

**Comment by paulidale** [comment]
Add an `enable-leading---for-enable` and then accept leading `-` for enable/disable options :)
